frontend/test/pytest/test_compiler.py

```diff
@@ -250,11 +250,9 @@ def workflow():
             qml.PauliX(wires=0)
             return qml.state()
 
-        directory = os.path.join(os.getcwd(), workflow.__name__)
-        files = os.listdir(directory)
+        files = os.listdir(str(workflow.workspace))
         # The directory is non-empty. Should at least contain the original .mlir file
         assert files
-        workflow.workspace.cleanup()
```

Comment:
Why is this removed? Don't we need to cleanup as long as `keep_intermediate=True`?

Reply:
I believe this test is not correct. The [workspace name calculation logic](https://github.com/PennyLaneAI/catalyst/blob/135ab94edb1415758aca497a8681096b7ff178c7/frontend/catalyst/utils/filesystem.py#L89) is more complicated than we expect here. I suggest to relax the conditions to just: 1) check that the workspace is _some_ directory and 2) it does contain some files

Reply:
Then this would keep polluting the environment if we ran the tests locally multiple times. That's what the cleanup is here for. I don't see how this test needs more relaxation, it has been passing for a long time in the CI/CD.

Reply:
Oh, I see. Let me just keep the cleanup. Hope you are fine with reading files from workspace rather than from `directory`.

Reply:
Done